app/src/main/kotlin/compiler/Compiler.kt

```diff
@@ -0,0 +1,10 @@
+package compiler
+
+import compiler.common.diagnostics.Diagnostics
+import java.io.Reader
+
+class Compiler(val diagnostics: Diagnostics) {
```

Comment:
I think we should use an interface for diagnostics here.
We can either replace `Diagnostics` with `(Diagnostic) -> Unit` here, or turn `Diagnostics` into a functional interface.
https://kotlinlang.org/docs/fun-interfaces.html